app/assets/javascripts/repositories/repository_datatable.js

```diff
@@ -32,27 +50,429 @@ var RepositoryDatatable = (function(global) {
       return value;
     });
 
-  var rowsSelected = [];
+  function changeToViewMode() {
+    currentMode = 'viewMode';
+    // Table specific stuff
+    TABLE.button(0).enable(true);
+    FilePreviewModal.init();
+  }
 
-  // Tells whether we're currently viewing or editing table
-  var currentMode = 'viewMode';
+  function changeToEditMode() {
+    currentMode = 'editMode';
+    // Table specific stuff
+    TABLE.button(0).enable(false);
+  }
 
-  // Tells what action will execute by pressing on save button (update/create)
-  var saveAction = 'update';
-  var selectedRecord;
+  // Enable/disable edit button
+  function updateButtons() {
+    if (currentMode === 'viewMode') {
+      $('#addRepositoryRecord').removeClass('disabled');
```

Comment:
Maybe we can combine objects by groups:
` $('#object1, #object2, ...').removeClass('disabled').prop('disabled', false)`